forgerock-auth/src/main/java/org/forgerock/android/auth/Config.java

```diff
@@ -99,28 +110,48 @@ public void setSharedPreferences(SharedPreferences sharedPreferences) {
         this.sharedPreferences = sharedPreferences;
     }
 
-    public synchronized void init(Context context) {
+    // We need to remove this in future, because the connected tests are depends on this. but the unit tests/integration tests are not depend on this.
+    @Deprecated
+    public synchronized void init(Context appContext) {
```

Comment:
From an architectural perspective I don't think that the Config class should be a singleton anymore. The fact that we need to have a reset() function that generates a new clean object, breaks slightly the whole pattern. I don't think we will create a memory leak or something so it's not urgent, but in my mind this Config should be hosted as instance property in another class ex: FRAuth that would act as the singleton.

Furthermore, now that we support dynamic configurations this class can be initialized multiple times in the app lifecycle which again breaks the singeton pattern.

Lastly, if someone calls the deprecated public synchronized void init(Context appContext) method and the class is alread initialized, the config will not change. I know that this is not the intented use of the API, but both functions (until the deprecated gets removed) should have the same outcome. There is one scenario that this might happen IRL. Someone starts the SDK by passing an FROptions object and then later calls the default init to re-start the SDK from the Srtings.xml configuration.